src/utilities/git.jl

```diff
@@ -1,5 +1,49 @@
 git_checkout(branch::AbstractString) = run(`git checkout $(branch)`)
 
+git_add(; items="", flags="") = run(`git add $flags $items`)
+
+git_remote_remove(remote::AbstractString) = run(`git remote remove $remote`)
+git_remote_add(remote::AbstractString, url::AbstractString) = run(`git remote add $remote $url`)
+
+git_reset(pathspec::AbstractString; flags="") = run(`git reset $flags "$pathspec"`)
+
+function git_push(remote::AbstractString, branch::AbstractString; force=false)
+    force_flag = force ? "-f" : ""
+
+    # TODO: Inject username/email here
```

Comment:
I don't think you need to add username/email when pushing only on commits as we want to tie them to a user?

Reply:
im not sure you can push without having the username/email set?